autoload/maktaba/syscall.vim

```diff
@@ -216,6 +302,30 @@ endfunction
 
 ""
 " @dict Syscall
+" Executes the system asynchronously and invokes the callback on completion.
+" [callback] function will be called on asynchronous command completion, with
```

Comment:
The callback arg doesn't look optional (and I wouldn't expect it to be). Can you update to use `{callback}` in the doc comments and `a:Callback` below? (Uppercase because funcrefs are valid and vimscript is weird.)

I'd also lean towards allow_sync_fallback being a required argument, because neither behavior is clearly a "reasonable default". Better if callers always consider which they want.